packages/typescript-estree/tests/lib/node-utils.test.ts

```diff
@@ -42,3 +45,30 @@ describe(unescapeStringLiteralText, () => {
     ).toBe(`a\n<>"'&©∆&rx;😂\u0000\u0001`);
   });
 });
+
+describe('unescapeUnicodeIdentifier', () => {
+  it('should decode simple \\uXXXX escape', () => {
+    expect(unescapeUnicodeIdentifier('\\u0061')).toBe('a');
+    expect(unescapeUnicodeIdentifier('\\u0042')).toBe('B');
+  });
+
+  it('should decode \\u{X} escape', () => {
+    expect(unescapeUnicodeIdentifier('\\u{61}')).toBe('a');
+    expect(unescapeUnicodeIdentifier('\\u{1F602}')).toBe('😂');
+  });
+
+  it('should decode multiple escapes in one string', () => {
+    expect(unescapeUnicodeIdentifier('\\u0061\\u0062')).toBe('ab');
+    expect(unescapeUnicodeIdentifier('\\u{61}\\u{62}')).toBe('ab');
+  });
+
+  it('should leave non-escape text unchanged', () => {
+    expect(unescapeUnicodeIdentifier('foo')).toBe('foo');
+    expect(unescapeUnicodeIdentifier('a\\u0062c')).toBe('abc');
+  });
+
+  it('should not decode invalid escapes', () => {
+    expect(unescapeUnicodeIdentifier('\\u00ZZ')).toBe('\\u00ZZ');
+    expect(unescapeUnicodeIdentifier('\\u{ZZ}')).toBe('\\u{ZZ}');
+  });
+});
```

Comment:
[Testing] Nit: I'm a fan of having one thing under test per `it()` case. I think it would make sense to split each of these two-paired ones into two `it()`s. But I wouldn't block on it, this file already doesn't align to that (and even more so in existing code!).